src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -456,7 +456,7 @@ protected override LocalSymbol MakeDeclarationExpressionVariable(DeclarationExpr
                             nodeBinder: _enclosingBinder,
                             typeSyntax: node.Type,
                             identifierToken: designation.Identifier,
-                            kind: LocalDeclarationKind.RegularVariable,
+                            kind: LocalDeclarationKind.DeclarationExpression,
```

Comment:
>kind: LocalDeclarationKind.DeclarationExpression, [](start = 28, length = 49)

It should be very easy to syntactically detect whether this is an Out Variable and use specific kind for that. I see no good reason to be specific for deconstruction variables and so imprecise for Out Variables. 
As an alternative, all of them (out/deconstruction/error) are based on SingleVariableDesignationSyntax. Why not use VariableDesignation kind for all of them? I believe the design should be consistent, either all get specific kinds, or none.